src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.yaml

```diff
@@ -97,6 +97,22 @@ spec:
     - field: Properties.autoUpgradeProfile.upgradeChannel
       notEquals: none
 
+---
+# Synopsis: Configure AKS clusters to use AAD pod identities to access Azure resources securely.
+apiVersion: github.com/microsoft/PSRule/v1
+kind: Rule
+metadata:
+  name: Azure.AKS.PodIdentity
+  tags:
+    release: 'preview'
+    ruleSet: '2021_12'
+spec:
+  type:
```

Comment:
@ArmaanMcleod Lets add a selector to only suggest this rule if Azure CNI is used. The latest guidance does not recommend this for kubenet clusters.

https://docs.microsoft.com/azure/aks/use-azure-ad-pod-identity#using-kubenet-network-plugin-with-azure-active-directory-pod-managed-identities

Selector example:

https://github.com/Azure/PSRule.Rules.Azure/blob/816c5c5086cd1c300f02781f9e8a952c9011bc10/src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.yaml#L44-L60

https://github.com/Azure/PSRule.Rules.Azure/blob/816c5c5086cd1c300f02781f9e8a952c9011bc10/src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.yaml#L104-L116